airbyte-db/db-lib/src/main/java/io/airbyte/db/factory/DataSourceFactory.java

```diff
@@ -196,19 +198,23 @@ private DataSourceBuilder() {}
      *
      * @param connectionProperties custom jdbc_url_parameters containing information on connection
      *        properties
+     * @param driverClassName name of the JDBC driver
      * @return DataSourceBuilder class used to create dynamic fields for DataSource
      */
-    private static long getConnectionTimeoutMs(final Map<String, String> connectionProperties) {
-      final Duration connectionTimeout;
-      // TODO: the usage of CONNECT_TIMEOUT_KEY is Postgres specific, may need to extend for other
-      // databases
-      connectionTimeout =
-          connectionProperties.containsKey(CONNECT_TIMEOUT_KEY) ? Duration.ofSeconds(Long.parseLong(connectionProperties.get(CONNECT_TIMEOUT_KEY)))
-              : CONNECT_TIMEOUT_DEFAULT;
-      if (connectionTimeout.getSeconds() == 0) {
-        return connectionTimeout.toMillis();
+    private static long getConnectionTimeoutMs(final Map<String, String> connectionProperties, String driverClassName) {
+      // TODO: the usage of CONNECT_TIMEOUT is Postgres specific, may need to extend for other databases
+
+      if (driverClassName.equals(DatabaseDriver.POSTGRESQL.getDriverClassName())) {
+        final String pgPropertyConnectTimeout = CONNECT_TIMEOUT.getName();
+        // If the PGProperty.CONNECT_TIMEOUT was set by the user, then take its value, if not take the
+        // default
+        return (connectionProperties.containsKey(pgPropertyConnectTimeout)
+            && (Long.parseLong(connectionProperties.get(pgPropertyConnectTimeout)) > 0))
+                ? Duration.ofSeconds(Long.parseLong(connectionProperties.get(pgPropertyConnectTimeout))).toMillis()
+                : Duration.ofSeconds(Long.parseLong(Objects.requireNonNull(CONNECT_TIMEOUT.getDefaultValue()))).toMillis();
```

Comment:
nit : can you make this a if/else condition instead of a ternary operator? I find that it's more readable that way

Reply:
> nit : can you make this a if/else condition instead of a ternary operator? I find that it's more readable that way

done